src/preact/utils/xhr.js

```diff
@@ -1,10 +1,25 @@
 export const xhrUtils = {
   /**
-   * Simply calls `fetch(url).then(r => r.json())`
+   * Wrapper around `fetch` that returns the parsed JSON result
+   *
+   * @param {RequestInfo} url
+   * @return {Promise<*>}
+   */
+  async fetchJson(url) {
+    return (await this.fetch(url)).json();
+  },
+
+  /**
+   * Wrapper around `fetch` that checks the status code
+   *
    * @param {RequestInfo} url
    * @return {Promise<Response>}
    */
-  fetchJson(url) {
-    return self.fetch(url).then((r) => r.json());
+  async fetch(url) {
```

Comment:
Should this maybe accept an optional `init` object to mirror https://developer.mozilla.org/en-US/docs/Web/API/fetch ?

Reply:
Also are we able to use async/await in code now? Typically we haven't been able to, but I do know a lot of infrastructure has been updated recently

Reply:
If needed in the future, yes.  There will be some improvements to these services in https://github.com/ampproject/amphtml/issues/37465, I will follow up with adding that param.